### PR TITLE
Harfanglab: improve parser

### PR DESCRIPTION
**Pull request by squioc**
- Fix the location of hashes (not process hashes but the hashes of the executable)
- Upgrade the parser to the current Harfanglab parser
- add support for registry keys and powershell command

### CODE REVIEW

**Review by Darkheir — DISMISSED** [review]

File: HarfangLab/harfanglab/_meta/fields.yml [hunk 1]
```diff
@@ -3,29 +3,44 @@
...
   name: harfanglab.alert_subtype
   type: keyword
-  description: ""
+  description: "The subtype of the alert"
 
 harfanglab.execution:
   name: harfanglab.execution
   type: long
-  description: ""
+  description: "duration of the execution"
```
Comment (PierrickV) [inline HarfangLab/harfanglab/_meta/fields.yml:21]:
```suggestion
  description: "Execution time "
```

File: HarfangLab/harfanglab/tests/alert.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
...alves\\\\Desktop\",\"pe_timestamp\":\"2021-04-30T07:37:03.000Z\",\"parent_unique_id\":\"ab376ee3-ecb4-4537-06a4-0031a7bc3d13\",\"pe_imphash\":\"B6BC868F5046CE7764D0627266AF0200\",\"hashes\":{\"md5\":\"46f3fa15de36a0825df4dbfd94614566\",\"sha256\":\"79dd65171cdbc3a1505d11921babf62ffcab4d725ab4a28813c02a6f3a2760ff\",\"sha1\":\"41453d9b0ae75f687c7720373d7586bc0a9d1607\"},\"size\":1430016,\"matched_rules\":[{\"namespace\":\"mimikatz.yar\",\"rulename\":\"mimikatz_clear_string_markers\",\"source\":\"default\"},{\"namespace\":\"mimikatz.yar\",\"rulename\":\"mimikatz_patches_x64\",\"source\":\"default\"}],\"create_time\":\"2021/04/30 13:20:54.315189\",\"parent_commandline\":\"\\\"C:\\\\Windows\\\\System32\\\\WindowsPowerShell\\\\v1.0\\\\powershell.exe\\\" \",\"rule_revision\":81,\"signed\":false,\"ppid\":1700,\"status\":0},\"alert_time\":\"2021-04-30T13:21:20.704+00:00\",\"alert_subtype\":\"process\",\"execution\":0,\"alert_type\":\"yara\",\"rule_id\":\"YARA binary check\",\"@version\":\"1\",\"@timestamp\":\"2021-04-30T13:21:20.717Z\",\"@event_create_date\":\"2021-04-30T13:21:20.704Z\",\"log_type\":\"alert\",\"status\":\"new\",\"alert_unique_id\":\"bd13fef8-a8fd-4d28-ad97-928a83b96085\"}",
```
Comment (PierrickV) [inline HarfangLab/harfanglab/tests/alert.json:56]:
```suggestion
```

We don't need to include the `expected["message"]` (improve readability)

File: HarfangLab/harfanglab/tests/alert_false_positive.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
...     \"internal_name\": \"tasklist.exe\",\n      \"product_name\": \"Microsoft® Windows® Operating System\"\n    },\n    \"session\": 3,\n    \"pe_timestamp\": \"1994-09-11T16:43:21.000Z\",\n    \"parent_unique_id\": \"2332edf8-70c0-43c2-4590-00f912ab3d18\",\n    \"process_name\": \"tasklist.exe\",\n    \"grandparent_commandline\": \"C:\\\\Program Files (x86)\\\\EPOS\\\\EPOS Connect\\\\EPOSConnect.exe 1\",\n    \"pe_timestamp_int\": 779301801,\n    \"grandparent_image\": \"C:\\\\Program Files (x86)\\\\EPOS\\\\EPOS Connect\\\\EPOSConnect.exe\",\n    \"parent_commandline\": \"C:\\\\WINDOWS\\\\system32\\\\cmd.exe /d /s /c tasklist\",\n    \"fake_parent_unique_id\": null,\n    \"size\": 79360,\n    \"fake_ppid\": null,\n    \"hashes\": {\n      \"sha1\": \"7f50d8c3cf3ec79122a876e969bdb65d939becd0\",\n      \"sha256\": \"76eac7b5f53e0d58a98d5a6ddf9c97e19d1462ef65c0035d7798f89988b15ab4\",\n      \"md5\": \"0a4448b31ce7f83cb7691a2657f330f1\"\n    }\n  },\n  \"execution\": 0,\n  \"rule_name\": \"Discovery: Process list\",\n  \"maturity\": \"stable\",\n  \"msg\": \"Detects the execution of tasklist.exe, a tool used to gather detailed information about a computer's active processes.\"\n}\n",
```
Comment (PierrickV) [inline HarfangLab/harfanglab/tests/alert_false_positive.json:12]:
```suggestion
```

File: HarfangLab/harfanglab/tests/network2.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    },
+    "host": {
+      "domain": "WORKGROUP",
+      "hostname": "REDACTED",
+      "os": {
+        "full": "Windows Server 2019 Datacenter",
+        "version": "10.0.17763"
+      }
+    },
+    "message": "{\"daddr\":\"172.31.9.222\",\"event_id\":3,\"dport\":3389,\"agent\":{\"osversion\":\"10.0.17763\",\"domainname\":\"WORKGROUP\",\"agentid\":\"d4e3bf36-929d-4ddf-8526-492e89955808\",\"osproducttype\":\"Windows Server 2019 Datacenter\",\"hostname\":\"REDACTED\",\"domain\":null},\"conn_type\":0,\"pid\":1004,\"sport\":42221,\"is_ipv6\":\"false\",\"username\":\"NT AUTHORITY\\\\NETWORK SERVICE\",\"@timestamp\":\"2021-05-01T09:55:44.688Z\",\"@version\":\"1\",\"image_name\":\"C:\\\\Windows\\\\System32\\\\svchost.exe\",\"saddr\":\"185.202.2.238\",\"direction\":\"in\",\"initiated\":\"false\",\"log_type\":\"network\",\"@event_create_date\":\"2021-05-01T09:55:30.372Z\",\"process_unique_id\":\"9aad140d-929d-4ddf-03ec-008c68ef14a0\"}",
```
Comment (PierrickV) [inline HarfangLab/harfanglab/tests/network2.json:39]:
```suggestion
```

File: HarfangLab/harfanglab/tests/wineventlog3.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
... ID\":\"0\",\"Action Name\":\"Non applicable\",\"Unused6\":null,\"Additional Actions String\":\"No additional actions required\",\"Threat ID\":\"2147772194\",\"Execution Name\":\"Suspendu\",\"Severity ID\":\"5\",\"Error Description\":\"L’opération a réussi. \",\"Engine Version\":\"AM: 1.1.18800.4, NIS: 1.1.18800.4\",\"Unused3\":null,\"Detection ID\":\"{2E51DC7F-A01D-4E9E-94C8-782C63D85C6E}\",\"Security intelligence Version\":\"AV: 1.355.1292.0, AS: 1.355.1292.0, NIS: 1.355.1292.0\",\"Product Name\":\"Antivirus Microsoft Defender\",\"Unused\":null,\"Action ID\":\"9\",\"Source ID\":\"3\",\"Additional Actions ID\":\"0\",\"Source Name\":\"Protection en temps réel\",\"Unused5\":null,\"Severity Name\":\"Grave\",\"Error Code\":\"0x00000000\",\"Unused2\":null},\"level\":\"Warning\",\"source_name\":\"Microsoft-Windows-Windows Defender\",\"opcode\":null,\"@timestamp\":\"2022-01-03T05:45:12.816Z\",\"user_data\":{},\"agent\":{\"agentid\":\"06d70013-58c7-46e3-9231-452a383af90b\",\"domain\":null,\"osproducttype\":\"Windows Server 2019 Datacenter\",\"ostype\":\"windows\",\"osversion\":\"10.0.17763\",\"version\":\"2.12.2\",\"domainname\":\"REDACTED\",\"distroid\":null,\"hostname\":\"REDACTED\"}}",
```
Comment (PierrickV) [inline HarfangLab/harfanglab/tests/wineventlog3.json:24]:
```suggestion
```

**Review by PierrickV — COMMENTED** [review]
> The subtype of the alert
> The timestamp of the alert


In english, the adjective comes after the name e.g. Alert subtype

**Review by PierrickV — APPROVED** [review]

**Pull request merged**